File: binding/src/network/libqabstractnetworkcache.cpp
#include <QAbstractNetworkCache>
#include <QDateTime>
#include <QHttpHeaders>
#include <QIODevice>
#include <QNetworkCacheMetaData>
#include <QObject>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <QUrl>
#include <qabstractnetworkcache.h>
#include "libqabstractnetworkcache.h"
#include "libqabstractnetworkcache.hxx"

QNetworkCacheMetaData* QNetworkCacheMetaData_new() {
	 return new QNetworkCacheMetaData();
}

QNetworkCacheMetaData* QNetworkCacheMetaData_new2(const QNetworkCacheMetaData* other) {
	 return new QNetworkCacheMetaData(*other);
}

void QNetworkCacheMetaData_OperatorAssign(QNetworkCacheMetaData* self, const QNetworkCacheMetaData* other) {
	self->operator=(*other);
}

void QNetworkCacheMetaData_Swap(QNetworkCacheMetaData* self, QNetworkCacheMetaData* other) {
	self->swap(*other);
}

bool QNetworkCacheMetaData_OperatorEqual(const QNetworkCacheMetaData* self, const QNetworkCacheMetaData* other) {
	return self->operator==(*other);
}

bool QNetworkCacheMetaData_OperatorNotEqual(const QNetworkCacheMetaData* self, const QNetworkCacheMetaData* other) {
	return self->operator!=(*other);
}

bool QNetworkCacheMetaData_IsValid(const QNetworkCacheMetaData* self) {
	return self->isValid();
}

QUrl* QNetworkCacheMetaData_Url(const QNetworkCacheMetaData* self) {
	return new QUrl(self->url());
}

void QNetworkCacheMetaData_SetUrl(QNetworkCacheMetaData* self, const QUrl* url) {
	self->setUrl(*url);
}

libqt_list QNetworkCacheMetaData_RawHeaders(const QNetworkCacheMetaData* self) {
	return self->rawHeaders();
}

void QNetworkCacheMetaData_SetRawHeaders(QNetworkCacheMetaData* self, const libqt_list headers) {
	self->setRawHeaders(*headers);
}

QHttpHeaders* QNetworkCacheMetaData_Headers(const QNetworkCacheMetaData* self) {
	return new QHttpHeaders(self->headers());
}

void QNetworkCacheMetaData_SetHeaders(QNetworkCacheMetaData* self, const QHttpHeaders* headers) {
	self->setHeaders(*headers);
}

QDateTime* QNetworkCacheMetaData_LastModified(const QNetworkCacheMetaData* self) {
	return new QDateTime(self->lastModified());
}

void QNetworkCacheMetaData_SetLastModified(QNetworkCacheMetaData* self, const QDateTime* dateTime) {
	self->setLastModified(*dateTime);
}

QDateTime* QNetworkCacheMetaData_ExpirationDate(const QNetworkCacheMetaData* self) {
	return new QDateTime(self->expirationDate());
}

void QNetworkCacheMetaData_SetExpirationDate(QNetworkCacheMetaData* self, const QDateTime* dateTime) {
	self->setExpirationDate(*dateTime);
}

bool QNetworkCacheMetaData_SaveToDisk(const QNetworkCacheMetaData* self) {
	return self->saveToDisk();
}

void QNetworkCacheMetaData_SetSaveToDisk(QNetworkCacheMetaData* self, bool allow) {
	self->setSaveToDisk(allow);
}

libqt_map QNetworkCacheMetaData_Attributes(const QNetworkCacheMetaData* self) {
	return self->attributes();
}

void QNetworkCacheMetaData_SetAttributes(QNetworkCacheMetaData* self, const libqt_map attributes) {
	self->setAttributes(*attributes);
}

void QNetworkCacheMetaData_Delete(QNetworkCacheMetaData* self) {
    delete self;
}

libqt_string QAbstractNetworkCache_Tr(const char* s) {
	QString _ret = QAbstractNetworkCache::tr(s);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

QNetworkCacheMetaData* QAbstractNetworkCache_MetaData(QAbstractNetworkCache* self, const QUrl* url) {
	return new QNetworkCacheMetaData(self->metaData(*url));
}

void QAbstractNetworkCache_UpdateMetaData(QAbstractNetworkCache* self, const QNetworkCacheMetaData* metaData) {
	self->updateMetaData(*metaData);
}

QIODevice* QAbstractNetworkCache_Data(QAbstractNetworkCache* self, const QUrl* url) {
	return self->data(*url);
}

bool QAbstractNetworkCache_Remove(QAbstractNetworkCache* self, const QUrl* url) {
	return self->remove(*url);
}

long long QAbstractNetworkCache_CacheSize(const QAbstractNetworkCache* self) {
	return self->cacheSize();
}

QIODevice* QAbstractNetworkCache_Prepare(QAbstractNetworkCache* self, const QNetworkCacheMetaData* metaData) {
	return self->prepare(*metaData);
}

void QAbstractNetworkCache_Insert(QAbstractNetworkCache* self, QIODevice* device) {
	self->insert(device);
}

void QAbstractNetworkCache_Clear(QAbstractNetworkCache* self) {
	self->clear();
}

libqt_string QAbstractNetworkCache_Tr2(const char* s, const char* c) {
	QString _ret = QAbstractNetworkCache::tr(s, c);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

libqt_string QAbstractNetworkCache_Tr3(const char* s, const char* c, int n) {
	QString _ret = QAbstractNetworkCache::tr(s, c, n);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

// Base class handler implementation
QNetworkCacheMetaData* QAbstractNetworkCache_QBaseMetaData(QAbstractNetworkCache* self, const QUrl* url) {
	auto* vqabstractnetworkcache = dynamic_cast<VirtualQAbstractNetworkCache*>(self);
	if (vqabstractnetworkcache && vqabstractnetworkcache->isVirtualQAbstractNetworkCache) {
vqabstractnetworkcache->setQAbstractNetworkCache_MetaData_IsBase(true);
	return new QNetworkCacheMetaData(vqabstractnetworkcache->metaData(*url));
}
}

// Auxiliary method to allow providing re-implementation
void QAbstractNetworkCache_OnMetaData(QAbstractNetworkCache* self, intptr_t slot) {
	auto* vqabstractnetworkcache = dynamic_cast<VirtualQAbstractNetworkCache*>(self);
	if (vqabstractnetworkcache && vqabstractnetworkcache->isVirtualQAbstractNetworkCache) {
vqabstractnetworkcache->setQAbstractNetworkCache_MetaData_Callback(reinterpret_cast<VirtualQAbstractNetworkCache::QAbstractNetworkCache_MetaData_Callback>(slot));
}
}

// Base class handler implementation
void QAbstractNetworkCache_QBaseUpdateMetaData(QAbstractNetworkCache* self, const QNetworkCacheMetaData* metaData) {
	auto* vqabstractnetworkcache = dynamic_cast<VirtualQAbstractNetworkCache*>(self);
	if (vqabstractnetworkcache && vqabstractnetworkcache->isVirtualQAbstractNetworkCache) {
vqabstractnetworkcache->setQAbstractNetworkCache_UpdateMetaData_IsBase(true);
	vqabstractnetworkcache->updateMetaData(*metaData);
}
}

// Auxiliary method to allow providing re-implementation
void QAbstractNetworkCache_OnUpdateMetaData(QAbstractNetworkCache* self, intptr_t slot) {
	auto* vqabstractnetworkcache = dynamic_cast<VirtualQAbstractNetworkCache*>(self);
	if (vqabstractnetworkcache && vqabstractnetworkcache->isVirtualQAbstractNetworkCache) {
vqabstractnetworkcache->setQAbstractNetworkCache_UpdateMetaData_Callback(reinterpret_cast<VirtualQAbstractNetworkCache::QAbstractNetworkCache_UpdateMetaData_Callback>(slot));
}
}

// Base class handler implementation
QIODevice* QAbstractNetworkCache_QBaseData(QAbstractNetworkCache* self, const QUrl* url) {
	auto* vqabstractnetworkcache = dynamic_cast<VirtualQAbstractNetworkCache*>(self);
	if (vqabstractnetworkcache && vqabstractnetworkcache->isVirtualQAbstractNetworkCache) {
vqabstractnetworkcache->setQAbstractNetworkCache_Data_IsBase(true);
	return vqabstractnetworkcache->data(*url);
}
}

// Auxiliary method to allow providing re-implementation
void QAbstractNetworkCache_OnData(QAbstractNetworkCache* self, intptr_t slot) {
	auto* vqabstractnetworkcache = dynamic_cast<VirtualQAbstractNetworkCache*>(self);
	if (vqabstractnetworkcache && vqabstractnetworkcache->isVirtualQAbstractNetworkCache) {
vqabstractnetworkcache->setQAbstractNetworkCache_Data_Callback(reinterpret_cast<VirtualQAbstractNetworkCache::QAbstractNetworkCache_Data_Callback>(slot));
}
}

// Base class handler implementation
bool QAbstractNetworkCache_QBaseRemove(QAbstractNetworkCache* self, const QUrl* url) {
	auto* vqabstractnetworkcache = dynamic_cast<VirtualQAbstractNetworkCache*>(self);
	if (vqabstractnetworkcache && vqabstractnetworkcache->isVirtualQAbstractNetworkCache) {
vqabstractnetworkcache->setQAbstractNetworkCache_Remove_IsBase(true);
	return vqabstractnetworkcache->remove(*url);
}
}

// Auxiliary method to allow providing re-implementation
void QAbstractNetworkCache_OnRemove(QAbstractNetworkCache* self, intptr_t slot) {
	auto* vqabstractnetworkcache = dynamic_cast<VirtualQAbstractNetworkCache*>(self);
	if (vqabstractnetworkcache && vqabstractnetworkcache->isVirtualQAbstractNetworkCache) {
vqabstractnetworkcache->setQAbstractNetworkCache_Remove_Callback(reinterpret_cast<VirtualQAbstractNetworkCache::QAbstractNetworkCache_Remove_Callback>(slot));
}
}

// Base class handler implementation
long long QAbstractNetworkCache_QBaseCacheSize(const QAbstractNetworkCache* self) {
	auto* vqabstractnetworkcache = dynamic_cast<const VirtualQAbstractNetworkCache*>(self);
	if (vqabstractnetworkcache && vqabstractnetworkcache->isVirtualQAbstractNetworkCache) {
vqabstractnetworkcache->setQAbstractNetworkCache_CacheSize_IsBase(true);
	return vqabstractnetworkcache->cacheSize();
}
}

// Auxiliary method to allow providing re-implementation
void QAbstractNetworkCache_OnCacheSize(const QAbstractNetworkCache* self, intptr_t slot) {
	auto* vqabstractnetworkcache = dynamic_cast<const VirtualQAbstractNetworkCache*>(self);
	if (vqabstractnetworkcache && vqabstractnetworkcache->isVirtualQAbstractNetworkCache) {
vqabstractnetworkcache->setQAbstractNetworkCache_CacheSize_Callback(reinterpret_cast<VirtualQAbstractNetworkCache::QAbstractNetworkCache_CacheSize_Callback>(slot));
}
}

// Base class handler implementation
QIODevice* QAbstractNetworkCache_QBasePrepare(QAbstractNetworkCache* self, const QNetworkCacheMetaData* metaData) {
	auto* vqabstractnetworkcache = dynamic_cast<VirtualQAbstractNetworkCache*>(self);
	if (vqabstractnetworkcache && vqabstractnetworkcache->isVirtualQAbstractNetworkCache) {
vqabstractnetworkcache->setQAbstractNetworkCache_Prepare_IsBase(true);
	return vqabstractnetworkcache->prepare(*metaData);
}
}

// Auxiliary method to allow providing re-implementation
void QAbstractNetworkCache_OnPrepare(QAbstractNetworkCache* self, intptr_t slot) {
	auto* vqabstractnetworkcache = dynamic_cast<VirtualQAbstractNetworkCache*>(self);
	if (vqabstractnetworkcache && vqabstractnetworkcache->isVirtualQAbstractNetworkCache) {
vqabstractnetworkcache->setQAbstractNetworkCache_Prepare_Callback(reinterpret_cast<VirtualQAbstractNetworkCache::QAbstractNetworkCache_Prepare_Callback>(slot));
}
}

// Base class handler implementation
void QAbstractNetworkCache_QBaseInsert(QAbstractNetworkCache* self, QIODevice* device) {
	auto* vqabstractnetworkcache = dynamic_cast<VirtualQAbstractNetworkCache*>(self);
	if (vqabstractnetworkcache && vqabstractnetworkcache->isVirtualQAbstractNetworkCache) {
vqabstractnetworkcache->setQAbstractNetworkCache_Insert_IsBase(true);
	vqabstractnetworkcache->insert(device);
}
}

// Auxiliary method to allow providing re-implementation
void QAbstractNetworkCache_OnInsert(QAbstractNetworkCache* self, intptr_t slot) {
	auto* vqabstractnetworkcache = dynamic_cast<VirtualQAbstractNetworkCache*>(self);
	if (vqabstractnetworkcache && vqabstractnetworkcache->isVirtualQAbstractNetworkCache) {
vqabstractnetworkcache->setQAbstractNetworkCache_Insert_Callback(reinterpret_cast<VirtualQAbstractNetworkCache::QAbstractNetworkCache_Insert_Callback>(slot));
}
}

// Base class handler implementation
void QAbstractNetworkCache_QBaseClear(QAbstractNetworkCache* self) {
	auto* vqabstractnetworkcache = dynamic_cast<VirtualQAbstractNetworkCache*>(self);
	if (vqabstractnetworkcache && vqabstractnetworkcache->isVirtualQAbstractNetworkCache) {
vqabstractnetworkcache->setQAbstractNetworkCache_Clear_IsBase(true);
	vqabstractnetworkcache->clear();
}
}

// Auxiliary method to allow providing re-implementation
void QAbstractNetworkCache_OnClear(QAbstractNetworkCache* self, intptr_t slot) {
	auto* vqabstractnetworkcache = dynamic_cast<VirtualQAbstractNetworkCache*>(self);
	if (vqabstractnetworkcache && vqabstractnetworkcache->isVirtualQAbstractNetworkCache) {
vqabstractnetworkcache->setQAbstractNetworkCache_Clear_Callback(reinterpret_cast<VirtualQAbstractNetworkCache::QAbstractNetworkCache_Clear_Callback>(slot));
}
}

void QAbstractNetworkCache_Delete(QAbstractNetworkCache* self) {
    delete self;
}

File: binding/include/network/libqabstractnetworkcache.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qnetworkcachemetadata.html
pub const qnetworkcachemetadata = struct {

    /// New constructs a new QNetworkCacheMetaData object.
    pub fn New() QtC.QNetworkCacheMetaData {
        return qtc.QNetworkCacheMetaData_new();
    }


    /// New2 constructs a new QNetworkCacheMetaData object.
    pub fn New2(other: ?*anyopaque) QtC.QNetworkCacheMetaData {
        return qtc.QNetworkCacheMetaData_new2(@ptrCast(other));
    }


    pub fn OperatorAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QNetworkCacheMetaData_OperatorAssign(@ptrCast(self), @ptrCast(other));
    }

    pub fn Swap(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QNetworkCacheMetaData_Swap(@ptrCast(self), @ptrCast(other));
    }

    pub fn OperatorEqual(self: ?*anyopaque, other: ?*anyopaque) bool {
        return qtc.QNetworkCacheMetaData_OperatorEqual(@ptrCast(self), @ptrCast(other));
    }

    pub fn OperatorNotEqual(self: ?*anyopaque, other: ?*anyopaque) bool {
        return qtc.QNetworkCacheMetaData_OperatorNotEqual(@ptrCast(self), @ptrCast(other));
    }

    pub fn IsValid(self: ?*anyopaque, ) bool {
        return qtc.QNetworkCacheMetaData_IsValid(@ptrCast(self));
    }

    pub fn Url(self: ?*anyopaque, ) QtC.QUrl {
        return qtc.QNetworkCacheMetaData_Url(@ptrCast(self));
    }

    pub fn SetUrl(self: ?*anyopaque, url: ?*anyopaque) void {
        qtc.QNetworkCacheMetaData_SetUrl(@ptrCast(self), @ptrCast(url));
    }

    pub fn RawHeaders(self: ?*anyopaque, ) []const u8 {
        return qtc.QNetworkCacheMetaData_RawHeaders(@ptrCast(self));
    }

    pub fn SetRawHeaders(self: ?*anyopaque, headers: []const u8) void {
        qtc.QNetworkCacheMetaData_SetRawHeaders(@ptrCast(self), @ptrCast(headers));
    }

    pub fn Headers(self: ?*anyopaque, ) QtC.QHttpHeaders {
        return qtc.QNetworkCacheMetaData_Headers(@ptrCast(self));
    }

    pub fn SetHeaders(self: ?*anyopaque, headers: ?*anyopaque) void {
        qtc.QNetworkCacheMetaData_SetHeaders(@ptrCast(self), @ptrCast(headers));
    }

    pub fn LastModified(self: ?*anyopaque, ) QtC.QDateTime {
        return qtc.QNetworkCacheMetaData_LastModified(@ptrCast(self));
    }

    pub fn SetLastModified(self: ?*anyopaque, dateTime: ?*anyopaque) void {
        qtc.QNetworkCacheMetaData_SetLastModified(@ptrCast(self), @ptrCast(dateTime));
    }

    pub fn ExpirationDate(self: ?*anyopaque, ) QtC.QDateTime {
        return qtc.QNetworkCacheMetaData_ExpirationDate(@ptrCast(self));
    }

    pub fn SetExpirationDate(self: ?*anyopaque, dateTime: ?*anyopaque) void {
        qtc.QNetworkCacheMetaData_SetExpirationDate(@ptrCast(self), @ptrCast(dateTime));
    }

    pub fn SaveToDisk(self: ?*anyopaque, ) bool {
        return qtc.QNetworkCacheMetaData_SaveToDisk(@ptrCast(self));
    }

    pub fn SetSaveToDisk(self: ?*anyopaque, allow: bool) void {
        qtc.QNetworkCacheMetaData_SetSaveToDisk(@ptrCast(self), allow);
    }

    pub fn Attributes(self: ?*anyopaque, ) QHash<QNetworkRequest::Attribute, QVariant> {
        return qtc.QNetworkCacheMetaData_Attributes(@ptrCast(self));
    }

    pub fn SetAttributes(self: ?*anyopaque, attributes: *const QHash<QNetworkRequest::Attribute, QVariant>) void {
        qtc.QNetworkCacheMetaData_SetAttributes(@ptrCast(self), @ptrCast(attributes));
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QNetworkCacheMetaData_Delete(@ptrCast(self));
    }
};

/// https://doc.qt.io/qt-6/qabstractnetworkcache.html
pub const qabstractnetworkcache = struct {

    pub fn Tr(s: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QAbstractNetworkCache_Tr(@ptrCast(s));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qabstractnetworkcache.Tr: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn MetaData(self: ?*anyopaque, url: ?*anyopaque) QtC.QNetworkCacheMetaData {
        return qtc.QAbstractNetworkCache_MetaData(@ptrCast(self), @ptrCast(url));
    }

    /// Allows for overriding the related default method
    pub fn OnMetaData(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) QtC.QNetworkCacheMetaData) void {
        qtc.QAbstractNetworkCache_OnMetaData(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseMetaData(self: ?*anyopaque, url: ?*anyopaque) QtC.QNetworkCacheMetaData {
        return qtc.QAbstractNetworkCache_QBaseMetaData(@ptrCast(self), @ptrCast(url));
    }

    pub fn UpdateMetaData(self: ?*anyopaque, metaData: ?*anyopaque) void {
        qtc.QAbstractNetworkCache_UpdateMetaData(@ptrCast(self), @ptrCast(metaData));
    }

    /// Allows for overriding the related default method
    pub fn OnUpdateMetaData(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QAbstractNetworkCache_OnUpdateMetaData(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseUpdateMetaData(self: ?*anyopaque, metaData: ?*anyopaque) void {
        qtc.QAbstractNetworkCache_QBaseUpdateMetaData(@ptrCast(self), @ptrCast(metaData));
    }

    pub fn Data(self: ?*anyopaque, url: ?*anyopaque) QtC.QIODevice {
        return qtc.QAbstractNetworkCache_Data(@ptrCast(self), @ptrCast(url));
    }

    /// Allows for overriding the related default method
    pub fn OnData(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) QtC.QIODevice) void {
        qtc.QAbstractNetworkCache_OnData(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseData(self: ?*anyopaque, url: ?*anyopaque) QtC.QIODevice {
        return qtc.QAbstractNetworkCache_QBaseData(@ptrCast(self), @ptrCast(url));
    }

    pub fn Remove(self: ?*anyopaque, url: ?*anyopaque) bool {
        return qtc.QAbstractNetworkCache_Remove(@ptrCast(self), @ptrCast(url));
    }

    /// Allows for overriding the related default method
    pub fn OnRemove(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) bool) void {
        qtc.QAbstractNetworkCache_OnRemove(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseRemove(self: ?*anyopaque, url: ?*anyopaque) bool {
        return qtc.QAbstractNetworkCache_QBaseRemove(@ptrCast(self), @ptrCast(url));
    }

    pub fn CacheSize(self: ?*anyopaque, ) i64 {
        return qtc.QAbstractNetworkCache_CacheSize(@ptrCast(self));
    }

    /// Allows for overriding the related default method
    pub fn OnCacheSize(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) i64) void {
        qtc.QAbstractNetworkCache_OnCacheSize(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseCacheSize(self: ?*anyopaque, ) i64 {
        return qtc.QAbstractNetworkCache_QBaseCacheSize(@ptrCast(self));
    }

    pub fn Prepare(self: ?*anyopaque, metaData: ?*anyopaque) QtC.QIODevice {
        return qtc.QAbstractNetworkCache_Prepare(@ptrCast(self), @ptrCast(metaData));
    }

    /// Allows for overriding the related default method
    pub fn OnPrepare(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) QtC.QIODevice) void {
        qtc.QAbstractNetworkCache_OnPrepare(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBasePrepare(self: ?*anyopaque, metaData: ?*anyopaque) QtC.QIODevice {
        return qtc.QAbstractNetworkCache_QBasePrepare(@ptrCast(self), @ptrCast(metaData));
    }

    pub fn Insert(self: ?*anyopaque, device: ?*anyopaque) void {
        qtc.QAbstractNetworkCache_Insert(@ptrCast(self), @ptrCast(device));
    }

    /// Allows for overriding the related default method
    pub fn OnInsert(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QAbstractNetworkCache_OnInsert(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseInsert(self: ?*anyopaque, device: ?*anyopaque) void {
        qtc.QAbstractNetworkCache_QBaseInsert(@ptrCast(self), @ptrCast(device));
    }

    pub fn Clear(self: ?*anyopaque, ) void {
        qtc.QAbstractNetworkCache_Clear(@ptrCast(self));
    }

    /// Allows for overriding the related default method
    pub fn OnClear(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QAbstractNetworkCache_OnClear(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseClear(self: ?*anyopaque, ) void {
        qtc.QAbstractNetworkCache_QBaseClear(@ptrCast(self));
    }

    pub fn Tr2(s: []const u8, c: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QAbstractNetworkCache_Tr2(@ptrCast(s), @ptrCast(c));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qabstractnetworkcache.Tr2: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Tr3(s: []const u8, c: []const u8, n: i32, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QAbstractNetworkCache_Tr3(@ptrCast(s), @ptrCast(c), n);
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qabstractnetworkcache.Tr3: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QAbstractNetworkCache_Delete(@ptrCast(self));
    }
};
